scripts/validate/shellcheck

```diff
@@ -1,11 +1,26 @@
 #!/usr/bin/env bash
-set -e
 
 # Maintain an array of files to shellcheck not the best solution but will do for the time being
 FILES=()
 FILES+=("contrib/completion/bash/docker")
+FILES+=("scripts/build/.variables")
+FILES+=("scripts/build/binary")
+FILES+=("scripts/build/cross")
+FILES+=("scripts/build/dynbinary")
+FILES+=("scripts/build/osx")
+FILES+=("scripts/build/windows")
+FILES+=("scripts/docs/generate-man.sh")
+FILES+=("scripts/docs/generate-yaml.sh")
+FILES+=("scripts/test/unit")
+FILES+=("scripts/test/unit-with-coverage")
+FILES+=("scripts/test/watch")
+FILES+=("scripts/validate/check-git-diff")
 FILES+=("scripts/validate/shellcheck")
 
+RETURN="0"
 for f in "${FILES[@]}"; do
     shellcheck "$f"
+    RETURN=$((RETURN+$?))
```

Comment:
I think this script could be a lot smaller:

```bash
 #!/usr/bin/env bash
set -eu -o pipefail

shellcheck contrib/completion/bash/docker
find scripts/ -type f | grep -v scripts/winresources | xargs shellcheck
```

I think this has the following advantages

* we should be able to remove a bunch of the `shellcheck disable=SC1091` above the `source` because the shellcheck run has access to all the files. This might actually improve shellcheck coverage, not sure.
* as new files are added to `scripts/` we don't have to remember to add them here, because I'm sure we wont remember
* the intent is more obvious (we're running shellcheck on two sets of files)
* we don't need to remove the `set -e`, it's fine if we fail after the first run
* we could potentially run each with different `--exclude` if we ever wanted to exclude a code from one type of script